docs/adr/0002-schema-first-api-design.md

```diff
@@ -0,0 +1,59 @@
+# ADR 0002: Schema-First API Design
+
+## Status
+Voorstel
+
+## Context
+Om op een consistente en duidelijke manier een goed onderhoudbare contract voor de BRP API te kunnen ontwerpen en te kunnen documenteren, is er behoefte aan een werkwijze die dit faciliteert.
+
+## Beslissing
+We hebben besloten om het contract voor de BRP API te documenteren met behulp van de [OpenAPI Specificatie][OAS] en de [Schema-First API Design][SchemaFirst] aanpak te hanteren voor de BRP API ontwikkeling.
+Dit betekent dat voor elk te implementeren wijziging/toevoeging eerst wordt gekeken of de implementatie wijziging/toevoeging zal leiden tot een wijziging van het API contract. De wijziging aan het API contract wordt dan vóór de implementatie wijziging/toevoeging uitgevoerd.
+
+## Consequenties
+- **Voordelen:**
+    - Duidelijke en consistente API contract.
+    - Verbeterde communicatie tussen de stakeholders.
+    - Eerder en gemakkelijker valideren van het API contract.
+    - Betere Developer Experience (betere documentatie, client code generatie en consumers en provider van het API contract kunnen parallel worden geïmplementeerd).
+
+- **Nadelen:**
+    - Mogelijke leercurve voor stakeholders die niet bekend zijn met Schema-First API design.
+
+## Overwogen alternatieven
+- **Code-First API Design:** Eerst de wijziging implementeren en vervolgens het API contract genereren vanuit de code. Deze aanpak is niet gekozen omdat de kans groot is dat deze aanpak zal leiden tot onduidelijke en inconsistente API contracten en verkeerde aannames tijdens de implementatie.
+
+## Afspraken
+
+- Een externe API contract (wordt geconsumeerd door applicaties van zowel binnen als buiten de organisatie van de provider) wordt beheerd in een voor dat API opgezette GitHub repo. Een voorbeeld van zo'n repo is de [Haal Centraal BRP bevragen][ApiContractRepo] repo. De benodigde OAS bestanden van/voor het API contract en het gepubliceerde (in genereervariant map) OAS bestand van het API contract staan in de **specificatie** map van de repo.
```

Comment:
Ik mis de argumentatie waarom externe API contracten in een aparte GitHub repository beheerd moeten worden, terwijl interne API contracten kunnen blijven in dezelfde GitHub repository waar de provider implementatie zich bevindt.

Reply:
@FrozenSync ik denk dat dat is omdat deze interne repo's verborgen zijn voor de buitenwereld. Interne interfaces maken geen deel uit van het product van RvIG. Het te publiceren, te beveiligen en te onderhouden product voor afnemers is zo klein mogelijk.